### PR TITLE
Allow `<Map>` component comparison to support configurable datasets/layers

### PR DESCRIPTION
**Pull request by ram-sharan25**
Previously, comparing datasets involved manually assigning the dataset to the properties of a dataset layer. With this change, you can now pass two additional parameters  `compareDatasetId` and `compareLayerId` to the `Map` block, allowing it to compare the specified dataset with the one defined by `datasetId` and `layerId`.

### CODE REVIEW

File: app/scripts/components/common/blocks/block-map.tsx [hunk 1]
```diff
@@ -112,6 +112,8 @@ interface MapBlockProps {
   datasets: VedaDatum<DatasetData>;
   dateTime?: string;
   compareDateTime?: string;
+  compareDataSetId?: string;
+  compareLayerId?: string;
```
Comment (sandrahoang686) [inline app/scripts/components/common/blocks/block-map.tsx:116]: nit: `compareDatasetId`, no need to capitalize the `s` but do we need to pass that in? It wont be used it looks like. datasetId is never referenced from `baseMapStaticCompareData`

File: app/scripts/components/common/blocks/block-map.tsx [hunk 2]
```diff
@@ -166,7 +170,13 @@ function MapBlock(props: MapBlockProps) {
   const layersToFetch = useMemo(() => {
     const [baseMapStaticData] = reconcileDatasets([layerId], datasetLayers, []);
     let totalLayers = [baseMapStaticData];
-    const baseMapStaticCompareData = baseMapStaticData.data.compare;
+    const baseMapStaticCompareData =
+      !!compareDataSetId && !!compareLayerId
```
Comment (sandrahoang686) [inline app/scripts/components/common/blocks/block-map.tsx:174]: nit: `!!(compareDatasetId && compareLayerId)`

**Review by sandrahoang686 — CHANGES_REQUESTED** [review]

File: app/scripts/components/common/blocks/block-map.tsx (hunk 1 above)
Comment (ram-sharan25) [reply app/scripts/components/common/blocks/block-map.tsx:116]: It is optional. We have to pass tha in ,if only we want to compare the datasets.

**Comment by hanbyul-here** [comment]
👋 This change needs a bit more thinking through regarding labeling. ex. We are depending on `compare` attribute of the base dataset to get the formatted label dynamically. If we are passing datasetId and layerId directly, we can't really use base dataset's label function. 
What is the expected label when compare datasetid and compare layer id are passed?

**Comment by slesaad** [comment]
> 👋 This change needs a bit more thinking through regarding labeling. ex. We are depending on `compare` attribute of the base dataset to get the formatted label dynamically. If we are passing datasetId and layerId directly, we can't really use base dataset's compare label function. What is the expected label when compare datasetid and compare layer id are passed?

I'd say a combination of dataset name, layer name, and date? @hanbyul-here

**Comment by aboydnw** [comment]
@slesaad is this needed for GHG in the near term, or is it more about pushing a useful feature that you all used on GHG back into veda-ui? 

If it's more about making a useful feature available in veda-ui, I'd advocate for making a ticket that can more holistically address the map comparison feature in stories. we haven't really touched on that in a while so it's probably due for an update soon. if it's more something that you need for GHG, maybe we push this forward as a quick n dirty solution, and then address more holistically later.

**Comment by slesaad** [comment]
@aboydnw for sure, please create a ticket. there's no rush to get it in GHG right now, but it's something we will need soon enough. didn't have all the details when this PR was created :D

**Pull request closed without merging**